### ISSUE TITLE
CoreOS/docker privileges: "operation not permitted"

### ISSUE THREAD

**Issue by xnaveira** [google/metallb#21]
Deploying the controller and the speaker on CoreOS servers running kubernetes the dockers never get to "Running" state. Examining the logs:

```
panic: standard_init_linux.go:178: exec user process caused "operation not permitted" [recovered]
	panic: standard_init_linux.go:178: exec user process caused "operation not permitted"

goroutine 1 [running, locked to thread]:
panic(0x7eb2e0, 0xc820142fc0)
	/usr/lib/go1.6/src/runtime/panic.go:481 +0x3e6
github.com/urfave/cli.HandleAction.func1(0xc8200b92f8)
	/build/amd64-usr/var/tmp/portage/app-emulation/runc-1.0.0_rc2_p9/work/runc-1.0.0_rc2_p9/Godeps/_workspace/src/github.com/urfave/cli/app.go:478 +0x38e
panic(0x7eb2e0, 0xc820142fc0)
	/usr/lib/go1.6/src/runtime/panic.go:443 +0x4e9
github.com/opencontainers/runc/libcontainer.(*LinuxFactory).StartInitialization.func1(0xc8200b8c08, 0xc820022098, 0xc8200b8d18)
	/build/amd64-usr/var/tmp/portage/app-emulation/runc-1.0.0_rc2_p9/work/runc-1.0.0_rc2_p9/Godeps/_workspace/src/github.com/opencontainers/runc/libcontainer/factory_linux.go:259 +0x136
github.com/opencontainers/runc/libcontainer.(*LinuxFactory).StartInitialization(0xc820060870, 0x7ff358a6e488, 0xc820142fc0)
	/build/amd64-usr/var/tmp/portage/app-emulation/runc-1.0.0_rc2_p9/work/runc-1.0.0_rc2_p9/Godeps/_workspace/src/github.com/opencontainers/runc/libcontainer/factory_linux.go:277 +0x5b1
main.glob.func8(0xc820080780, 0x0, 0x0)
	/build/amd64-usr/var/tmp/portage/app-emulation/runc-1.0.0_rc2_p9/work/runc-1.0.0_rc2_p9/main_unix.go:26 +0x68
reflect.Value.call(0x74fac0, 0x9012a0, 0x13, 0x847808, 0x4, 0xc8200b9278, 0x1, 0x1, 0x0, 0x0, ...)
	/usr/lib/go1.6/src/reflect/value.go:435 +0x120d
reflect.Value.Call(0x74fac0, 0x9012a0, 0x13, 0xc8200b9278, 0x1, 0x1, 0x0, 0x0, 0x0)
	/usr/lib/go1.6/src/reflect/value.go:303 +0xb1
github.com/urfave/cli.HandleAction(0x74fac0, 0x9012a0, 0xc820080780, 0x0, 0x0)
	/build/amd64-usr/var/tmp/portage/app-emulation/runc-1.0.0_rc2_p9/work/runc-1.0.0_rc2_p9/Godeps/_workspace/src/github.com/urfave/cli/app.go:487 +0x2ee
github.com/urfave/cli.Command.Run(0x84a6b8, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x8e05e0, 0x51, 0x0, ...)
	/build/amd64-usr/var/tmp/portage/app-emulation/runc-1.0.0_rc2_p9/work/runc-1.0.0_rc2_p9/Godeps/_workspace/src/github.com/urfave/cli/command.go:191 +0xfec
github.com/urfave/cli.(*App).Run(0xc820001980, 0xc82000a100, 0x2, 0x2, 0x0, 0x0)
	/build/amd64-usr/var/tmp/portage/app-emulation/runc-1.0.0_rc2_p9/work/runc-1.0.0_rc2_p9/Godeps/_workspace/src/github.com/urfave/cli/app.go:240 +0xaa4
main.main()
	/build/amd64-usr/var/tmp/portage/app-emulation/runc-1.0.0_rc2_p9/work/runc-1.0.0_rc2_p9/main.go:137 +0xe24
```

Removing ```allowPrivilegeEscalation: false``` from the manifests for both the deployment and the daemonset eliminates the problem.

OS: CoreOS 1520.6.0
Kubernetes: 1.8.4
Docker: Docker version 1.12.6, build a82d35e

**Comment by danderson** [comment]
?! Wow, that's bizarre. I wonder what CoreOS is doing that minikube and Ubuntu (my personal "prod" cluster) are not. I guess I need to spin up a CoreOS test cluster.

I've patched out `allowPrivilegeEscalation: false` from the manifests until I figure out what CoreOS is doing differently.

Thank you for the bug report! And sorry you're having such a rough time testing things :(

**Comment by danderson** [comment]
Hmm, looking at the traceback, it seems like it's runC itself that is crashing, not even MetalLB... So either runC itself is not compatible with `allowPrivilegeEscalation: false`, or something about how I'm creating/installing the binary is violating a CoreOS-specific security policy.

Could you tell me the version of CoreOS (`cat /etc/os-release`) and Docker (`sudo docker version`) on your k8s node? I want to try and reproduce the environment as exactly as possible.

**Comment by xnaveira** [comment]
You have the versions at the end of the first post :)

I'm deploying the cluster using [Typhoon](https://github.com/poseidon/typhoon)

**Comment by danderson** [comment]
Doh, I suck at reading, obviously :)

And TIL Typhoon! Very nice, maybe I should use that for my bare metal cluster. Thanks for all the data, I'll compare versions and troubleshoot once I get home and can do A/B testing with my own cluster. I'll re-add `allowPrivilegeEscalation: false` once I've figured out what's going on.

**Comment by danderson** [comment]
moby/moby#27983 might be a related issue. It's a past bug in runc where certain configurations removed filesystem access capabilities from runc before it executes the container entrypoint. Looks like this was fixed a long time ago, so maybe this is a regression, or a weird configuration in Typhoon...

**Comment by danderson** [comment]
One possibility: many similar bugs seem to be caused by incorrect SELinux policies. CoreOS does not activate SELinux enforcement by default, but maybe Typhoon does?...

**Comment by xnaveira** [comment]
Good news! As i mentioned i was running CoreOS [1520.6.0](https://coreos.com/releases/#1520.6.0) which apparently still used Docker 1.12.6. As it happens just yesterday they released CoreOS [1576.4.0](https://coreos.com/releases/#1576.4.0) which is the first version to use Docker 17.09.0. I updated my cluster and now the containers are running just fine with ```allowPrivilegeEscalation: false```!

**Comment by SailingYYC** [comment]
This is also an issue on:

```shell
Container Linux by CoreOS stable (1911.4.0)

# docker info
Containers: 39
 Running: 21
 Paused: 0
 Stopped: 18
Images: 15
Server Version: 1.12.6
Storage Driver: overlay2
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: overlay null bridge host
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp selinux
Kernel Version: 4.14.81-coreos
Operating System: Container Linux by CoreOS 1911.4.0 (Rhyolite)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.65 GiB
Name: obscured
ID: 3FVM:M7P6:BQ4B:AG3S:DQF4:AFJS:EKSU:V4T4:BBF5:YDHL:7QIJ:U7YM
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
```
